Failing tests test_get_machine and test_get_machine_app_specific

While packaging this package (version 235) I get these two tests failing:
```
[   21s] =================================== FAILURES ===================================
[   21s] _______________________________ test_get_machine _______________________________
[   21s] 
[   21s]     def test_get_machine():
[   21s] >       u1 = id128.get_machine()
[   21s] E       FileNotFoundError: [Errno 2] No such file or directory
[   21s] 
[   21s] systemd/test/test_id128.py:24: FileNotFoundError
[   21s] ________________________ test_get_machine_app_specific _________________________
[   21s] 
[   21s]     def test_get_machine_app_specific():
[   21s]         a1 = uuid.uuid1()
[   21s]         a2 = uuid.uuid1()
[   21s]     
[   21s]         with skip_oserror(errno.ENOSYS):
[   21s] >           u1 = id128.get_machine_app_specific(a1)
[   21s] E           FileNotFoundError: [Errno 2] No such file or directory
[   21s] 
[   21s] systemd/test/test_id128.py:33: FileNotFoundError
[   21s] =========================== short test summary info ============================
[   21s] FAILED systemd/test/test_id128.py::test_get_machine - FileNotFoundError: [Err...
[   21s] FAILED systemd/test/test_id128.py::test_get_machine_app_specific - FileNotFou...
[   21s] ============ 2 failed, 64 passed, 2 skipped, 1 deselected in 0.29s =============
```
[Complete build log](https://github.com/systemd/python-systemd/files/9356800/_log.txt) with versions of all packages used and steps taken to reproduce.

FileNotFoundError could possibly be influenced by the limited building environment we build our packages in (something from `/dev/` or something like this?). Or are there some real files missing?

Hmm, the second one seems to be gone, but the former still remains and I have discovered we have had skipped test_reader_this_machine as well.
```
[   23s] =================================== FAILURES ===================================
[   23s] _______________________________ test_get_machine _______________________________
[   23s] 
[   23s]     def test_get_machine():
[   23s] >       u1 = id128.get_machine()
[   23s] E       FileNotFoundError: [Errno 2] No such file or directory
[   23s] 
[   23s] systemd/test/test_id128.py:24: FileNotFoundError
[   23s] ___________________________ test_reader_this_machine ___________________________
[   23s] 
[   23s] tmpdir = local('/tmp/pytest-of-abuild/pytest-0/test_reader_this_machine0')
[   23s] 
[   23s]     def test_reader_this_machine(tmpdir):
[   23s]         j = journal.Reader(path=tmpdir.strpath)
[   23s]         with j:
[   23s] >           j.this_machine()
[   23s] 
[   23s] systemd/test/test_journal.py:236: 
[   23s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[   23s] 
[   23s] self = <systemd.journal.Reader object at 0x7fcb7f1dae00>, machineid = None
[   23s] 
[   23s]     def this_machine(self, machineid=None):
[   23s]         """Add match for _MACHINE_ID equal to the ID of this machine.
[   23s]     
[   23s]         If specified, machineid should be either a UUID or a 32 digit hex
[   23s]         number.
[   23s]     
[   23s]         Equivalent to add_match(_MACHINE_ID='machineid').
[   23s]         """
[   23s]         if machineid is None:
[   23s] >           machineid = _id128.get_machine().hex
[   23s] E           FileNotFoundError: [Errno 2] No such file or directory
[   23s] 
[   23s] systemd/journal.py:396: FileNotFoundError
[   23s] =========================== short test summary info ============================
[   23s] FAILED systemd/test/test_id128.py::test_get_machine - FileNotFoundError: [Err...
[   23s] FAILED systemd/test/test_journal.py::test_reader_this_machine - FileNotFoundE...
[   23s] =================== 2 failed, 64 passed, 3 skipped in 0.28s ====================
```

And yes, we don’t have `/etc/machine-id` in our build environment.

Again, [complete build log](https://github.com/systemd/python-systemd/files/9358078/_log.txt)

--- a/.github/workflows/install.yml
+++ b/.github/workflows/install.yml
@@ -76,6 +76,4 @@ jobs:
           # Avoid importing the systemd module from the git repository
           cd /
           python3 -c 'from systemd import journal; print(journal.__version__)'
-          # Initialize /etc/machine-id if it's empty to make the tests happy
-          [[ ! -s /etc/machine-id ]] && systemd-id128 new >/etc/machine-id
           pytest -v --pyargs systemd

--- a/systemd/test/test_id128.py
+++ b/systemd/test/test_id128.py
@@ -6,11 +6,11 @@ import pytest
 from systemd import id128
 
 @contextlib.contextmanager
-def skip_oserror(code):
+def skip_oserror(*errnos):
     try:
         yield
     except (OSError, IOError) as e:
-        if e.errno == code:
+        if e.errno in errnos:
             pytest.skip()
         raise
 
@@ -21,7 +21,10 @@ def test_randomize():
     assert u1 != u2
 
 def test_get_machine():
-    u1 = id128.get_machine()
+    # yikes, python2 doesn't know ENOMEDIUM
+    with skip_oserror(errno.ENOENT, errno.ENOSYS, 123):
+        u1 = id128.get_machine()
+
     u2 = id128.get_machine()
     assert u1 == u2
 
@@ -29,7 +32,7 @@ def test_get_machine_app_specific():
     a1 = uuid.uuid1()
     a2 = uuid.uuid1()
 
-    with skip_oserror(errno.ENOSYS):
+    with skip_oserror(errno.ENOENT, errno.ENOSYS, 123):
         u1 = id128.get_machine_app_specific(a1)
 
     u2 = id128.get_machine_app_specific(a2)

--- a/systemd/test/test_journal.py
+++ b/systemd/test/test_journal.py
@@ -233,7 +233,11 @@ def test_reader_this_boot(tmpdir):
 def test_reader_this_machine(tmpdir):
     j = journal.Reader(path=tmpdir.strpath)
     with j:
-        j.this_machine()
+        try:
+            j.this_machine()
+        except OSError:
+            pass
+
         j.this_machine(TEST_MID)
         j.this_machine(TEST_MID.hex)
 

